csharp/Ice/Filesystem/Client/Program.cs

```diff
@@ -12,35 +12,36 @@
 
 // Recursively list the contents of the root directory.
 Console.WriteLine("Contents of root directory:");
-ListRecursive(rootDir);
+await ListRecursiveAsync(rootDir);
 
 /// <summary>Recursively print the contents of a directory in tree fashion. For files, show the contents of each file.
 /// </summary>
 /// <param name="dir">The directory to list./<param>
 /// <param name="depth">The current nesting level (for indentation).</param>
-void ListRecursive(DirectoryPrx dir, int depth = 0)
+async Task ListRecursiveAsync(DirectoryPrx dir, int depth = 0)
 {
     var indent = new string('\t', ++depth);
 
-    NodePrx?[] contents = dir.List();
+    NodePrx?[] contents = await dir.ListAsync();
 
     foreach (NodePrx? node in contents)
     {
         Debug.Assert(node is not null); // The node proxies returned by list() are never null.
 
+        // TODO: should be an async call
```

Comment:
I think we should generate a checkedCastAsync ... anything else would be inconsistent.

Reply:
Adding checkedCastAsync sounds good. On a side note the Slice model for this example isn't great.

I wondering if we should add a NodeInfo struct  and NodeKind enum:

```
enum NodeKind
{
   File,
   Directory
}

struct NodeInfo
{
    string name;
    NodeKind kind;
}

sequence<NodeInfo> NodeInfoSeq;

interface Directory extends Node
{
    idempotent NodeInfoSeq list();
}
```

This avoids the need for separate name, checkedCast requests.